### ISSUE TITLE
GitHub's free tier only allows 1 assignee per issue, causes problems for `reviews assign` with `--num-reviews > 1`

### ISSUE THREAD

**Issue by DanielEWeeks** [repobee/repobee#696]
When using RepoBee on a previously-deployed set of GitHub Classroom repositories, after running `review setup`, then `review assign` worked well in assigning two peer reviewers to review their fellow student's repository, setting up the review teams and giving read access to the review team. However, it only opened a peer review issue for the first student in the list of two peer reviewers - it did not open a peer review issue for the second student in the list.  

Command used:

```
repobee reviews assign -a 3-r-characters-homework --sf students.txt --num-reviews 2 --issue review_issue.md
```

**Comment by slarse** [comment]
`reviews assign` should open just the one issue per repo. Then it should assign all reviewers to that single issue. Did it not assign the reviewers correctly? The tests are passing, and I just ran a manual test that worked fine as well, so I've been unable to reproduce this.

If it didn't correctly assign students, could you provide me with the following info:

* The amount of students in `students.txt`
* Do you have your own GitHub Enterprise, or are you using github.com?
    - If self-hosted Enterprise, which version is it (should be visible by the GitHub symbol at the bottom of any web page on the instance)?

Thanks!

**Comment by DanielEWeeks** [comment]
Sorry, maybe I misunderstood what `review assign` was supposed to do.  It did open one issue per repository, and both of the two students assigned to do the peer review have read access to both of the two repositories that they are supposed to review.  

But I thought if a student went to their assigned 'Issues' list at `https://github.com/issues/assigned`, they would see two assigned issues, one for each of the two repositories they were assigned to review.  In the example in the RepoBee documentation, it says that "The review team students were assigned to the issue as well".  

If I search for assigned issues using a filter like this: `is:open is:issue assignee:daniel-e-weeks archived:false`, it seems that each student was the assignee for only one issue, not two.  Oh, but I see that GitHub.com appears to only allow one person per issue, as the Assignees option on an Issue says "Assign up to 1 person to this issue". 

The solution is simple on my end: Instead of telling students to go to their assigned Issues list, they just need to look in their list of repositories to find the two that they have read access to.  

Thank you.

**Comment by slarse** [comment]
> But I thought if a student went to their assigned 'Issues' list at https://github.com/issues/assigned, they would see two assigned issues, one for each of the two repositories they were assigned to review.

This is exactly what should happen.

>  Oh, but I see that GitHub.com appears to only allow one person per issue, as the Assignees option on an Issue says "Assign up to 1 person to this issue"

Ah, okay, this is the problem. RepoBee will try to assign two people to the issue, but since there is a limit of 1 assignee in your organization, only whan assignee will "stick". I was not aware there was any form of membership on GitHub that allows only 1 assignee on an issue, but it appears as if the free organization plan only allows this. I'm guessing this is the pricing plan you are using? You could then solve this by applying for a the [teacher's toolbox](https://education.github.com/toolbox), which will allow you to upgrade your organization to a Team for free (it's a pricing plan, not to be confused with teams within organization). A Team allows up to 10 assignees per issue.

GitHub did not allow private repos in free organizations when I implemented the reviews features, and so I was not aware of this restriction. As it happens, this problem is also pertinent on GitLab's free tier, which also only allows a single assignee per issue. Perhaps it would be best to simply open one issue per reviewer and assign them to it, as it would be best if the functionality works well even with the free tiers.

**Comment by slarse** [comment]
#532

**Comment by DanielEWeeks** [comment]
I had qualified for the Teacher's Toolbox, but apparently missed the step of upgrading this semester's GitHub Classroom organization.  Thank you for pointing that out - I have now done that.

**Comment by slarse** [comment]
Cool, so now the `reviews assign` command should work as expected (get back to me if it doesn't). I still find it troubling that it doesn't work correctly for the free tiers, so a redesign seems in order.

Thanks for reporting this!

**Comment by slarse** [comment]
This problem isn't solved yet, but #789 outlines the solution, so I'm closing this issue as it's not actionable outside of #789.
